modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/NodeJSServerCodegen.java

```diff
@@ -241,6 +249,46 @@ public String apiFileFolder() {
     }
 
     @Override
+    public void processOpts() {
+        super.processOpts();
+
+        if (additionalProperties.containsKey(GOOGLE_CLOUD_FUNCTIONS)) {
+            setGoogleCloudFunctions(
+                Boolean.valueOf(additionalProperties.get(GOOGLE_CLOUD_FUNCTIONS).toString()));
+        }
+
+        if (additionalProperties.containsKey(EXPORTED_NAME)) {
+            setExportedName((String)additionalProperties.get(EXPORTED_NAME));
+        }
+
+        /*
+         * Supporting Files.  You can write single files for the generator with the
+         * entire object tree available.  If the input file has a suffix of `.mustache
+         * it will be processed by the template engine.  Otherwise, it will be copied
+         */
+        // supportingFiles.add(new SupportingFile("controller.mustache",
+        //   "controllers",
+        //   "controller.js")
+        // );
+        supportingFiles.add(new SupportingFile("swagger.mustache",
+                        "api",
+                        "swagger.yaml")
+        );
+        if (getGoogleCloudFunctions()) {
+            writeOptional(outputFolder, new SupportingFile("index-gcf.mustache", "", "index.js"));
+        } else {
+            writeOptional(outputFolder, new SupportingFile("index.mustache", "", "index.js"));
+        }
+        writeOptional(outputFolder, new SupportingFile("package.mustache", "", "package.json"));
+        writeOptional(outputFolder, new SupportingFile("README.mustache", "", "README.md"));
+        if (System.getProperty("noservice") == null) {
```

Comment:
Actually this comes from the original NodeJSCodegen.java, I just moved the code location. I'm not adding new.

Also, the `noservice` is for a different purpose from the selective generation.

The nodejs-server codegen generates two files per API. For the example of petstore, it generates files like `controllers/Pet.js` and `controllers/PetService.js` (see https://github.com/swagger-api/swagger-codegen/tree/master/samples/server/petstore/nodejs/controllers).

When this `noservice` is specified, it omits the generation of `PetService.js` file or any other -Service.js files. Only `Pet.js` or those files.